Core/src/org/sleuthkit/autopsy/communications/MessageNode.java

```diff
@@ -18,23 +18,147 @@
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_EMAIL_FROM;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_EMAIL_TO;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_TO;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_SUBJECT;
+import static org.sleuthkit.datamodel.BlackboardAttribute.TSK_BLACKBOARD_ATTRIBUTE_VALUE_TYPE.DATETIME;
+import org.sleuthkit.datamodel.TimeUtilities;
+import org.sleuthkit.datamodel.TskCoreException;
 
 /**
- *
- * 
+ * Wraps a BlackboardArtifact as an AbstractNode for use in an OutlookView
  */
-final class MessageNode extends BlackboardArtifactNode {
+final class MessageNode extends AbstractNode {
     private static final Logger logger = Logger.getLogger(RelationshipNode.class.getName());
     
+    private final BlackboardArtifact artifact;
+    
     MessageNode(BlackboardArtifact artifact) {
-        super(artifact);
-        // Grabbed this from RelationshipNode, does always work even with internalization?
+        super(Children.LEAF, Lookups.fixed(artifact));
+        this.artifact = artifact;
+        
         final String stripEnd = StringUtils.stripEnd(artifact.getDisplayName(), "s"); // NON-NLS
         String removeEndIgnoreCase = StringUtils.removeEndIgnoreCase(stripEnd, "message"); // NON-NLS
         setDisplayName(removeEndIgnoreCase.isEmpty() ? stripEnd : removeEndIgnoreCase);
+       
+        int typeID = artifact.getArtifactTypeID();
+        
+        String filePath = "org/sleuthkit/autopsy/images/"; //NON-NLS
+        if( typeID == BlackboardArtifact.ARTIFACT_TYPE.TSK_EMAIL_MSG.getTypeID()) {
+            filePath = filePath + "mail-icon-16.png"; //NON-NLS
+        } else if (typeID == BlackboardArtifact.ARTIFACT_TYPE.TSK_MESSAGE.getTypeID()) {
+            filePath = filePath + "message.png"; //NON-NLS
+        } else if (typeID == BlackboardArtifact.ARTIFACT_TYPE.TSK_CALLLOG.getTypeID()) {
+            filePath = filePath + "calllog.png"; //NON-NLS
+        }
+        
+        setIconBaseWithExtension(filePath);
+    }
+    
+    @Override
+    protected Sheet createSheet() {
+        Sheet sheet = new Sheet();
+        Sheet.Set sheetSet = sheet.get(Sheet.PROPERTIES);
+        if (sheetSet == null) {
+            sheetSet = Sheet.createPropertiesSet();
+            sheet.put(sheetSet);
+        }
+
+        sheetSet.put(new NodeProperty<>("Type", "Type", "Type", getDisplayName())); //NON-NLS
+        
+        BlackboardArtifact.ARTIFACT_TYPE fromID = BlackboardArtifact.ARTIFACT_TYPE.fromID(artifact.getArtifactTypeID());
+        if (null != fromID) {
+            //Consider refactoring this to reduce boilerplate
+            switch (fromID) {
+                case TSK_EMAIL_MSG:
+                    sheetSet.put(new NodeProperty<>("From", "From", "From",
```

Comment:
These NodeProperty Strings should probably be declared using @Messages so that they can be localized when necessary since they are user facing column names.

Reply:
Of the three strings you supply to NodeProperty, the first string is an identifier that should not be internationalized, the second some is a display name for the property\column which I have moved the to messages, the third string is a description that we do not show any where so I just pass an empty string.